systemFlake.nix

```diff
@@ -26,24 +26,28 @@
 
 let
   evalHostArgs =
-    { channelName ? "nixpkgs"
+    { channelName ? null
     , modules ? []
-    , system ? defaultSystem
+    , system ? null
     , extraArgs ? {}
-    , ...
-    }: defaultHostAttrs
-      // { 
-        inherit channelName system; 
-        modules = sharedModules ++ modules;
-        extraArgs = sharedExtraArgs // extraArgs;
-      };
+    }: {
+      # hard coded fallbacks
+      channelName = "nixpkgs";
+      system = defaultSystem or "x86_64-linux";
+    }
+    //  defaultHostAttrs
+    // {
+      inherit channelName system; 
+      modules = sharedModules ++ (if defaultHostAttrs ? modules then defaultHostAttrs.modules else []) ++ modules;
+      extraArgs = sharedExtraArgs // (if defaultHostAttrs ? extraArgs then defaultHostAttrs.extraArgs else {}) // extraArgs;
```

Comment:
nix experts: a more nixonic way here would be?

Reply:
```suggestion
      modules = sharedModules ++ defaultHostAttrs.modules or [] ++ modules;
      extraArgs = sharedExtraArgs // defaultHostAttrs.extraArgs or {} // extraArgs;
```